### PR TITLE
chore(tracing): include events in the trace

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/server/trace/recorder/tracer.ts [hunk 1]
```diff
@@ -108,23 +103,45 @@ class ContextTracer {
     await this._snapshotter.start(false);
   }
 
-  async _captureSnapshot(name: 'before' | 'after' | 'action', sdkObject: SdkObject, metadata: CallMetadata, element?: ElementHandle): Promise<void> {
+  async _captureSnapshot(name: 'before' | 'after' | 'action' | 'event', sdkObject: SdkObject, metadata: CallMetadata, element?: ElementHandle): Promise<void> {
```
Comment (dgozman) [inline src/server/trace/recorder/tracer.ts:106]: Drop `async` from this method?

File: src/server/trace/viewer/traceModel.ts [hunk 2]
```diff
@@ -84,9 +84,8 @@ export class TraceModel {
       case 'action': {
         const metadata = event.metadata;
         const { pageEntry } = this.pageEntries.get(metadata.pageId!)!;
-        const actionId = event.contextId + '/' + metadata.pageId + '/' + pageEntry.actions.length;
         const action: ActionEntry = {
-          actionId,
+          actionId: metadata.id,
```
Comment (dgozman) [inline src/server/trace/viewer/traceModel.ts:88]: How does `actionById` method work now?

File: src/server/trace/recorder/tracer.ts [hunk 3]
```diff
@@ -108,23 +103,45 @@ class ContextTracer {
...
       type: 'action',
       contextId: this._contextId,
       metadata,
-      snapshots: snapshotsForMetadata(metadata),
+    };
+    this._appendTraceEvent(event);
+    this._pendingCalls.delete(metadata.id);
+  }
+
+  onEvent(sdkObject: SdkObject, metadata: CallMetadata) {
```
Comment (dgozman) [inline src/server/trace/recorder/tracer.ts:137]: We can now drop things like `trace.LoadEvent` and such.

**Review by dgozman — APPROVED** [review]

File: src/server/trace/viewer/traceModel.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/server/trace/viewer/traceModel.ts:88]: I think it is unused.

File: src/server/trace/recorder/tracer.ts (hunk 3 above)
Comment (pavelfeldman) [reply src/server/trace/recorder/tracer.ts:137]: Not yet, I don't yet render the events on timeline...

**Pull request merged**